realtime/websocket/processors.py

```diff
@@ -0,0 +1,125 @@
+                elif data.get("type") == "audio":
+                    audio_bytes = base64.b64decode(data.get("data"))
+                    audio_data += audio_bytes
+
+                    if len(audio_data) < 2:
+                        continue
+                    if len(audio_data) % 2 != 0:
+                        # TODO: Get audio dtype from the frontend instead of hardcoding it
+                        array = np.frombuffer(audio_data[:-1], dtype=np.int16).reshape(1, -1)
+                        audio_data = audio_data[-1:]
+                    else:
+                        # TODO: Get audio dtype from the frontend instead of hardcoding it
+                        array = np.frombuffer(audio_data, dtype=np.int16).reshape(1, -1)
+                        audio_data = b""
+
+                    frame = av.AudioFrame.from_ndarray(array, format="s16", layout="mono")
+                    frame.sample_rate = self.sample_rate
+                    await self.audio_output_stream.put(frame)
+            except Exception as e:
+                logging.error("websocket: Exception", e)
+                raise asyncio.CancelledError()
+
+
+class WebsocketOutputStream:
+    """
+    Handles outgoing WebSocket messages by streaming audio and text data.
+
+    Attributes:
+        ws (WebSocket): The WebSocket connection.
+    """
+
+    def __init__(self, ws: WebSocket):
+        self.ws = ws
+
+    async def run(
+        self, audio_stream: AudioStream, message_stream: TextStream, video_stream: VideoStream, byte_stream: ByteStream
+    ):
+        """
+        Starts tasks to process and send byte and text streams.
+
+        Args:
+            audio_stream (AudioStream): The audio stream to send.
+            message_stream (TextStream): The text stream to send.
+            video_stream (VideoStream): The video stream to send.
+            byte_stream (ByteStream): The byte stream to send.
+        """
+        # TODO: Implement video stream and audio stream processing
+        await asyncio.gather(self.task(byte_stream), self.task(message_stream))
+
+    async def task(self, input_stream):
+        """
+        Sends data from the input stream over the WebSocket.
+
+        Args:
+            input_stream (Stream): The stream from which to send data.
+        """
+        while True:
+            data = await input_stream.get()
+            if data is None:
+                json_data = {"type": "audio_end", "timestamp": time.time()}
+                await self.ws.send_json(json_data)
+            elif isinstance(data, bytes):
+                output_bytes_io = io.BytesIO()
+                in_memory_wav = wave.open(output_bytes_io, "wb")
+
+                # TODO: Get channels, sample width, and sample rate from TTS module instead of hardcoding it
+                in_memory_wav.setnchannels(1)
+                in_memory_wav.setsampwidth(2)
+                in_memory_wav.setframerate(16000)
```

Comment:
set the sample rate in __init__ as well and use it here. use the default value of 16000

Reply:
init of WebsocketOutputStream

Reply:
in newer version audio data will contain this data so shouldn't do it within init